drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#define DT_DRV_COMPAT nxp_irtc
```

Comment:
very minor nit: put the DT_DRV_COMPAT above the devicetree.h include and group all the includes together